test/unit/cpu/buildIndexSet.cxx

```diff
@@ -66,103 +67,109 @@ Index_type buildIndexSet(IndexSet* hindex, IndexSetBuildMethod build_method)
+
+    // This is already being done above as iset_master
     case AddSegments: {
-      for (size_t i = 0; i < segments.size(); ++i) {
-        hindex[build_method].push_back(*segments[i]);
+      // This is already being done above as iset_master
+      for (size_t i = 0; i < iset_master.getNumSegments(); ++i) {
+        iset_master.segment_push_into(i, hindex[build_method], PUSH_BACK, PUSH_COPY);
       }
-
       break;
     }
 
     case AddSegmentsReverse: {
-      int last_i = static_cast<int>(segments.size() - 1);
-      for (int i = last_i; i >= 0; --i) {
-        hindex[build_method].push_front(*segments[i]);
+      RAJA::IndexSet<RAJA::RangeSegment, RAJA::ListSegment, RAJA::RangeStrideSegment>& iset_master = hindex[0];
+      for (int i = iset_master.getNumSegments()-1; i >= 0; --i) {
+        iset_master.segment_push_into(i, hindex[build_method], PUSH_FRONT, PUSH_COPY);
       }
 
       break;
     }
 
     case AddSegmentsNoCopy: {
-      IndexSet& iset_master = hindex[0];
-
+      RAJA::IndexSet<RAJA::RangeSegment, RAJA::ListSegment, RAJA::RangeStrideSegment>& iset_master = hindex[0];
       for (size_t i = 0; i < iset_master.getNumSegments(); ++i) {
-        hindex[build_method].push_back_nocopy(iset_master.getSegment(i));
+        iset_master.segment_push_into(i, hindex[build_method], PUSH_BACK, PUSH_NOCOPY);
       }
 
       break;
     }
 
     case AddSegmentsNoCopyReverse: {
-      IndexSet& iset_master = hindex[0];
-
-      int last_i = static_cast<int>(iset_master.getNumSegments() - 1);
-      for (int i = last_i; i >= 0; --i) {
-        hindex[build_method].push_front_nocopy(iset_master.getSegment(i));
+      RAJA::IndexSet<RAJA::RangeSegment, RAJA::ListSegment, RAJA::RangeStrideSegment>& iset_master = hindex[0];
+      for ( int i = iset_master.getNumSegments() - 1; i >= 0 ; --i ) {
+        iset_master.segment_push_into(i, hindex[build_method], PUSH_FRONT, PUSH_NOCOPY);
       }
 
       break;
     }
 
-    case MakeViewRange: {
-      IndexSet& iset_master = hindex[0];
+    case MakeSliceRange: {
+      RAJA::IndexSet<RAJA::RangeSegment, RAJA::ListSegment, RAJA::RangeStrideSegment>& iset_master = hindex[0];
       size_t num_segs = iset_master.getNumSegments();
+      RAJA::IndexSet<RAJA::RangeSegment, RAJA::ListSegment, RAJA::RangeStrideSegment>* iset_slice
+        = iset_master.createSlice(0, num_segs);
 
-      IndexSet* iset_view = iset_master.createView(0, num_segs);
-
-      for (size_t i = 0; i < iset_view->getNumSegments(); ++i) {
-        hindex[build_method].push_back_nocopy(iset_view->getSegment(i));
+      for (size_t i = 0; i < iset_slice->getNumSegments(); ++i) {
+        iset_slice->segment_push_into(i, hindex[build_method], PUSH_BACK, PUSH_NOCOPY);
       }
 
       break;
     }
 
-    case MakeViewArray: {
-      IndexSet& iset_master = hindex[0];
+    case MakeSliceArray: {
+      RAJA::IndexSet<RAJA::RangeSegment, RAJA::ListSegment, RAJA::RangeStrideSegment>& iset_master = hindex[0];
```

Comment:
Line break? Please align args vertically.  There are more of these things. I'll stop commenting on them.